### DISCUSSION TITLE
Can't get OCI for helm dependencies working

### DISCUSSION THREAD

**Discussion by Yaytay** [argoproj/argo-cd#5915] [Q&A]
Hi,

Using ArgoCD 2.0.0-rc2 I have a helm Chart that has this in the Chart.yaml:
```
dependencies:
  - name: my-library
    version: 0.0.4
    repository: oci://registry.gitlab.com/my-group/devops/my-library
```

This works just fine locally - I can do `helm dependecies build` and it downloads the Chart correctly and then I can run the template.

Trying to run this on ArgoCD gives this error:
```
rpc error: code = Unknown desc = Manifest generation error (cached): `helm dependency build` failed exit status 1: Error: repository oci://registry.gitlab.com/group-gti/devops/gti-helm-template-library is an OCI registry: this feature has been marked as experimental and is not enabled by default. Please set HELM_EXPERIMENTAL_OCI=1 in your environment to use this feature
```

Our argocd-cm config map is created entirely by yaml in our app-of-apps repo.
I've tried three different ways to specify the Helm repo in that yaml:
1.  Not at all.
2. With a top level folder:
```
    - url: registry.gitlab.com/my-group
      type: helm
      enableOci: true
      passwordSecret: {name: gitlab-user, key: password}
      usernameSecret: {name: gitlab-user, key: username}
```
2. And with a specific folder:
```
    - url: registry.gitlab.com/my-group/devops/my-library
      type: helm
      enableOci: true
      passwordSecret: {name: gitlab-user, key: password}
      usernameSecret: {name: gitlab-user, key: username}
```

Whatever I do still gets stuck with ArgoCD not setting HELM_EXPERIMENTAL_OCI.

Should this work, and if so, what am I missing?

Thanks.

Jim

**Comment by alexmt** [comment]
Hello @Yaytay , this is definitely a bug. Working on the fix

**Reply by Yaytay** [reply]
Great, thanks.

If it's possible I'd really like to _not_ have to specify a repo for every helm package we have (given that the way GitLab works they will have very different paths beyond the registry.gitlab.com/my-group part.

**Reply by alexmt** [reply]
https://github.com/argoproj/argo-cd/pull/5917

**Reply by alexmt** [reply]
@Yaytay I had the same thought. To achieve that we would have to just always set `HELM_EXPERIMENTAL_OCI=1` . Probably this is safe but I'm hesitant to make this change in v2.0 release without careful testing. Can you please create ticket https://github.com/argoproj/argo-cd/issues/new/choose ?

**Reply by jannfis** [reply]
To be honest, I think if there's a reason for Helm to have users explicitly enable this feature, we should not enable it by default.

**Reply by Yaytay** [reply]
I'm still not able to get my OCI dependencies working with the v2.0.0 release.
Any idea where it's breaking?
